frontend/src/routes/_oh.tsx

```diff
@@ -204,6 +207,7 @@ export default function MainApp() {
 
   const handleEndSession = () => {
     setStartNewProjectModalIsOpen(false);
+    store.dispatch(setCurrentAgentState(AgentState.LOADING));
```

Comment:
Should use the `useDispatch` hook (`const dispatch = useDispatch()`) instead of directly updating the store.

Reply:
good call!